contracts/DefaultStrategy.sol

```diff
@@ -0,0 +1,517 @@
+     * @param group The address of the group to add to the set of votable
+     * groups.
+     * @dev Fails if the maximum number of groups are already being voted for by
+     * the Account smart contract (as per the `maxNumGroupsVotedFor` in the
+     * Election contract).
+     */
+    function activateGroup(address group) external onlyManager {
+        if (!groupHealth.isValidGroup(group)) {
+            revert GroupNotEligible(group);
+        }
+
+        if (IManager(manager).groupsContain(group)) {
+            revert GroupAlreadyAdded(group);
+        }
+
+        if (IManager(manager).deprecatedGroupsContain(group)) {
+            if (!IManager(manager).removeDeprecatedGroup(group)) {
+                revert FailedToRemoveDeprecatedGroup(group);
+            }
+        }
+
+        if (
+            IManager(manager).getGroupsLength() + IManager(manager).getDeprecatedGroupsLength() >=
+            getElection().maxNumGroupsVotedFor() &&
+            !getElection().allowedToVoteOverMaxNumberOfGroups(address(account))
+        ) {
+            revert MaxGroupsVotedForReached();
+        }
+    }
+
+    /**
+     * @notice Sorts an array of GroupWithVotes structs based on increasing
+     * `votes` values.
+     * @param groupsWithVotes The array to sort.
+     * @dev This is an in-place insertion sort. In general in Solidity we should
+     * be careful of algorithms on arrays, especially O(n^2) ones, but here
+     * we're guaranteed to be working with a small array, its length is bounded
+     * by the maximum number of groups that can be voted for in Elections.sol.
+     */
+    function sortGroupsWithVotes(GroupWithVotes[] memory groupsWithVotes) internal pure {
+        for (uint256 i = 1; i < groupsWithVotes.length; i++) {
+            uint256 j = i;
+            while (j > 0 && groupsWithVotes[j].votes < groupsWithVotes[j - 1].votes) {
+                (groupsWithVotes[j], groupsWithVotes[j - 1]) = (
+                    groupsWithVotes[j - 1],
+                    groupsWithVotes[j]
+                );
+                j--;
+            }
+        }
+    }
+
+    /**
+     * @notice Returns a list of group addresses with their corresponding
+     * current total votes, sorted by the number of votes, and the total number
+     * of votes in the system.
+     * @param groups The array of addresses of the groups to sort.
+     * @return The array of GroupWithVotes structs, sorted by number of votes.
+     * @return The total number of votes assigned to active groups.
+     */
+    function getSortedGroupsWithVotes(address[] memory groups)
+        internal
+        view
+        returns (GroupWithVotes[] memory, uint256)
+    {
+        GroupWithVotes[] memory groupsWithVotes = new GroupWithVotes[](groups.length);
+        uint256 totalVotes = 0;
+        for (uint256 i = 0; i < groups.length; i++) {
+            uint256 votes;
+            if (specificGroupStrategy.isSpecificGroupStrategy(groups[i])) {
```

Comment:
Haven't quite looked deep through all this logic, but why do we care about the SpecificGroupStrategy votes for the group here?

Reply:
Because it is possible that we have one validator group that is used as active group but also as specific strategy.

In such case it is necessary to ignore funds from specificGroupStrategy in Defautl strategy.